### ISSUE TITLE
[A11Y] Registration form date picker activation button has no label

### ISSUE THREAD

**Issue by foxbunny** [indico/indico#6060]
**Describe the bug**

The date picker activation button contains an icon that is not announced, making the purpose of the button unclear.

**To Reproduce**
1. Go to a registration form that includes a date picker using a screen reader
2. Tab to the activation button
3. Observe that the button is announced as just "button"

**Expected behavior**

Expected the button to be announced as "activate date picker" or something along those lines.

**Additional context**
https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html

**Comment by foxbunny** [comment]
This is not fixable on our end. The button label is hard-coded upstream and not customizable.
